src/libraries/Common/src/System/Net/NTAuthentication.Common.cs

```diff
@@ -297,12 +302,17 @@ internal int MakeSignature(byte[] buffer, int offset, int count, [AllowNull] ref
                 SSPIHandleCache.CacheCredential(_credentialsHandle);
             }
 
+            byte[]? result =
+                resultBlobLength == 0 || _tokenBuffer == null ? null :
+                (_tokenBuffer.Length == resultBlobLength ? _tokenBuffer : _tokenBuffer.AsSpan(0, resultBlobLength).ToArray());
```

Comment:
One of the few places where slicing an array into an array is actually desired, so this could use that functionality we've exposed:
```suggestion
            byte[]? result =
                resultBlobLength == 0 || _tokenBuffer == null ? null :
                _tokenBuffer.Length == resultBlobLength ? _tokenBuffer :
                _tokenBuffer[0..resultBlobLength];
```